.github/workflows/snap.yaml

```diff
@@ -0,0 +1,145 @@
+name: Build Spark client snap and run tests
+
+env:
+  VERSION: 0.1
+  RELEASE: edge
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  build:
+    name: Build Snap
+    runs-on: ubuntu-latest
+    outputs:
+      snap-file: ${{ steps.build-snap.outputs.snap }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: dev
+
+      - id: build-snap
+        name: Build snap
+        uses: snapcore/action-build@v1
+        with:
+          snapcraft-channel: 7.x/candidate
+
+      - name: Upload built snap job artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: spark_client_snap_amd64
+          path: spark_client_${{env.VERSION}}_amd64.snap
+
+  test:
+    name: Test Snap
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+      - name: Download snap file
+        uses: actions/download-artifact@v3
+        with:
+          name: spark_client_snap_amd64
+          path: .
+
+      - name: Install snap file
+        run: |
+          sudo snap install spark_client_${{env.VERSION}}_amd64.snap --dangerous
+
+      - name: Setup Spark client
+        run: |
+          # connect interfaces
+          sudo snap connect spark-client:enable-kubeconfig-access
+          sudo snap connect spark-client:enable-scala-history
+          
+          # setup pre requisites
+          sudo snap install microk8s --classic
+          sudo usermod -a -G microk8s $USER
+          mkdir -p ~/.kube
+          microk8s config > ~/.kube/config
+          sudo chown -f -R $USER ~/.kube
+          microk8s status --wait-ready
+          sudo snap alias microk8s.kubectl kubectl
+          
+          # create the service account
+          sudo snap run spark-client.setup-spark-k8s \
+          service-account \
+          --kubeconfig ~/.kube/config \
+          --cluster microk8s-cluster    \
+          canonical-spark  \
+          default
+          
+          # create the ca certificate
+          sudo snap run spark-client.setup-spark-k8s \
+          get-ca-cert \
+          --kubeconfig ~/.kube/config \
+          --cluster microk8s-cluster > ~/ca.crt
+          
+          # create the oauth token
+          sudo snap run spark-client.setup-spark-k8s \
+          get-token \
+          --kubeconfig ~/.kube/config \ 
+          --cluster microk8s-cluster \ 
+          `kubectl get secrets | grep canonical-spark-token | cut -d' ' -f1` \
+          > ~/token
+          
+
+      - name: Run example job
+        env:
+          DOCKER_USERNAME: ${{ secrets.DockerUserName }}
+          DOCKER_PASSWORD: ${{ secrets.DockerPassword }}
```

Comment:
Hi @paulomach , need your help to setup secrets {DockerUserName, DockerPassword} for this github repo.

Context - I am trying to build CI for the spark client snap. The test for the snap requires spark docker image to run. Secrets will be used for docker login.

Reply:
Cool! As our chat, since it's only pulling and the image is public you can dismiss the login here!